compiler/src/main/kotlin/edu/cornell/cs/apl/viaduct/codegeneration/PlainTextCodeGenerator.kt

```diff
@@ -0,0 +1,483 @@
+        sender: LetNode,
+        sendProtocol: Protocol,
+        receiveProtocol: Protocol,
+        events: ProtocolCommunication
+    ): CodeBlock {
+        val receiveBuilder = CodeBlock.builder()
+        if (sendProtocol != receiveProtocol) {
+            val projection = ProtocolProjection(receiveProtocol, receivingHost)
+            var cleartextInputs = events.getProjectionReceives(projection, Plaintext.INPUT)
+            val clearTextTemp = codeGeneratorContext.newTemporary("clearTextTemp")
+
+            // initialize cleartext receive value by receiving from first host
+            if (cleartextInputs.isNotEmpty()) {
+                receiveBuilder.addStatement(
+                    "val %N = %L",
+                    clearTextTemp,
+                    codeGeneratorContext.receive(
+                        typeTranslator(typeAnalysis.type(sender)),
+                        cleartextInputs.first().send.host
+                    )
+                )
+                cleartextInputs = cleartextInputs.minusElement(cleartextInputs.first())
+            }
+
+            // receive from the rest of the hosts and compare against clearTextValue
+            for (event in cleartextInputs) {
+
+                // check to make sure that you got the same data from all hosts
+                receiveBuilder.beginControlFlow(
+                    "if(%N != %L)",
+                    clearTextTemp,
+                    codeGeneratorContext.receive(
+                        typeTranslator(typeAnalysis.type(sender)),
+                        event.send.host
+                    )
+                )
+                receiveBuilder.addStatement(
+                    "throw %T(%S)",
+                    runtimeErrorClass,
+                    "Plaintext : received different values"
+                )
+                receiveBuilder.endControlFlow()
+            }
+
+            // calculate set of hosts with whom [receivingHost] needs to check for equivocation
+            var hostsToCheckWith: Set<Host> =
+                events
+                    .filter { event ->
+                        // remove events where receiving host is not receiving plaintext data
+                        event.recv.id == Plaintext.INPUT &&
+
+                            // remove events where a host is sending data to themselves
+                            event.send.host != event.recv.host &&
+
+                            // remove events where [receivingHost] is the sender of the data
+                            event.send.host != receivingHost
+                    }
+                    // of events matching above criteria, get set of data receivers
+                    .map { event -> event.recv.host }
+                    // remove [receivingHost] from the set of hosts with whom [receivingHost] needs to
+                    // check for equivocation
+                    .filter { host -> host != receivingHost }
+                    .toSet()
```

Comment:
You should sort this list for deterministic code generation. Also, do you need `.toSet`? If you just iterate through the whole thing, keep it a list.